components/mongodb/actions/search-documents/search-documents.mjs

```diff
@@ -0,0 +1,34 @@
+import mongodbApp from "../../mongodb.app.mjs";
+
+export default {
+  key: "mongodb-search-documents",
+  name: "Search Documents",
+  description: "Search for specific documents or return all documents. [See the docs here](https://docs.mongodb.com/manual/reference/method/db.collection.find/)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    mongodbApp,
+    collection: {
+      propDefinition: [
+        mongodbApp,
+        "collection",
+      ],
+      description: "Collection containing the document to search for.",
+    },
+    filter: {
+      label: "Filter",
+      description: "Object to use as a filter",
+      type: "object",
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const documents = await this.mongodbApp.searchDocuments(
+      this.collection,
+      this.filter,
+    );
```

Comment:
If you add props `parseNumbers`, `parseBoolean`, and `parseDates`, you can pass them into `searchDocuments`.
```suggestion
    const documents = await this.mongodbApp.searchDocuments(
      this.collection,
      this.filter,
      undefined,
      undefined,
      this.parseNumbers,
      this.parseBooleans,
      this.parseDates,
    );
```